Caltech-IPAC/firefly
# FIREFLY-56 Change ruler tool behavior for image with or without WCS and HiPS

The developement fixes the ruler tool behavior based on the following JIRA tickets, 
https://jira.ipac.caltech.edu/browse/FIREFLY-56
https://jira.ipac.caltech.edu/browse/FIREFLY-1

Here is the  summry of the fixing, 
- increase the line thickness and font size and style of the ruler tool
- display PA along angular distance without requiring the user to click in the layer dialog
   --  PA angle and angular disatance are displayed in parallell, above and along the anglular distance vector 
   -- PA angle is measured East from North, in unit 'degree'
- Add an arrow head at the end of angllar distance vector 
- Unit update including that on layer dialog, 
   -- add unit 'pixel' to image with WCS case 
   -- image without WCS has only unit 'pixel' option
   -- HiPS has no 'pixel' unit
-  lines and the relevant text from offset calculation
   -- add handles (squares) at both ends of the lines. 
   -- text is rotated per the direction of each line

test: 
https://irsawebdev9.ipac.caltech.edu/FIREFLY-56_RulerToolChange/firefly/
- start image search on regular image, image without WCS, or HiPS
   (samples for image without WCS are located at firefly_test_data/no-projection-images)
- turn on the distance tool icon and locate the the tool on the image. 
- start the layer dialog, click 'offset calculation' and select the unit to check the display of the tool. 
- for non HiPS case, rotate the image and view the distance tool is rotated accordingly.
